[SRCH-6115] remove access to features that we plan to discontinue.

luisgmetzger opened the pull request:
## Summary
- This PR removes several deprecated sections/components/partials that are no longer needed to be viewed by customer/admin.
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review. If you cannot complete an item below, replace the checkbox with the ⚠️ `:warning:` emoji and explain why the step was not completed.
 
#### Functionality Checks
 
- [x] You have merged the latest changes from the target branch (usually `main`) into your branch.
  
- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket.

- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release - SRCH-####, SRCH-####, SRCH-####** matching the Jira ticket numbers in the release.
 
- [x] Automated checks pass. If Code Climate checks do not pass, explain reason for failures:
 
#### Process Checks

- [x] You have specified at least one "Reviewer".

stevenbarragan commented on spec/lib/js_fetcher_spec.rb:
```diff
@@ -1,11 +0,0 @@
-# frozen_string_literal: true
-
-describe JsFetcher do
```
The file `lib/js_fetcher.rb` is still been used. Why removing the test? If it's failing why is it failing? I'm not comfortable with just deleting the test.

luisgmetzger replied:
@stevenbarragan - Here is a bit of investigation on how it is being used and why I removed it: https://github.com/GSA/search-gov/pull/1863#issuecomment-3229514366

luisgmetzger replied:
You will notice that the `base_url` for search, this: `https://search.gov` now redirects.

luisgmetzger replied:
alternatively, I could maybe update the cassette to try and use the new domain `https://digital.gov/guides/search` 
the only thing I don't know is if this html file needs to exist: `javascript-test.html`

stevenbarragan commented:
Maybe it's easier to mock the selenium webdriver and verify that it got called with such url

luisgmetzger commented:
> Maybe it's easier to mock the selenium webdriver and verify that it got called with such url

Sure, let me give that a go.

luisgmetzger commented:
> Maybe it's easier to mock the selenium webdriver and verify that it got called with such url

@stevenbarragan  - Here is a quick update on the `js_fetcher_spec` file where I am now mocking the driver: 

What I changed
- Removed the previous JsFetcher integration test that hit a live page (previously [https://search.gov/javascript-test.html](https://search.gov/javascript-test.html)).
- Rewrote the JsFetcher spec to be a true unit test by stubbing Selenium:
    - Mocked Selenium::WebDriver::Options.firefox and Selenium::WebDriver.for.
    - Verified the driver is created, navigates to the expected URL (driver.get(url)), returns page_source, and quits.
    - Stubbed sleep and timeouts to keep the test fast and deterministic.

- Removed the need for the old VCR cassette tied to that live page.
- Updated UrlStatusCodeFetcher specs to use [https://digital.gov/guides/search](https://digital.gov/guides/search) as the valid URL, avoiding the 301 redirect we were seeing on search.gov and expecting a 200 from the new docs location.

Why it works now
- The JsFetcher spec no longer depends on external network behavior or a specific page’s JS content, so it’s stable, fast, and won’t break due to redirects or content changes.
- Stubbing Selenium lets us precisely assert the behavior we care about (correct URL passed, driver lifecycle) without requiring a real browser (headless) in CI.
- Switching the UrlStatusCodeFetcher’s valid URL to digital.gov avoids the 301 that caused earlier failures, aligning expectations with the new host.

luisgmetzger commented on .circleci/config.yml:
```diff
@@ -22,6 +22,7 @@ executors:
       - image: cimg/ruby:<< parameters.ruby_version >>
         environment:
           RAILS_ENV: test
+          BUNDLER_VERSION: "2.6.3"
```
@stevenbarragan  - I actually was facing an issue earlier with the CI not recognizing bundler version. I took this off now and it works.

luisgmetzger commented on .ruby-version:
```diff
@@ -1 +1 @@
-ruby-3.3.7
+3.3.7
```
I recently changed my ruby version. Reverted back.

stevenbarragan reviewed (changes requested):
Circle Ci is passing just fine now. what's with the changes on the configuration?

luisgmetzger commented:
> Circle Ci is passing just fine now. what's with the changes on the configuration?

@stevenbarragan  - I have reverted the changes and it's all passing now.
